src/transaction/xdc.ts

```diff
@@ -0,0 +1,814 @@
+ *
+ * @param provider url of the XDC Server to connect to. If not set, default public server will be used.
+ * @param fromPrivateKey optional private key of sender account
+ */
+export const getXdcClient = (provider?: string, fromPrivateKey?: string) => {
+    const client = new Web3(provider || `${TATUM_API_URL}/v3/xdc/web3/${process.env.TATUM_API_KEY}`);
+    if (fromPrivateKey) {
+        client.eth.accounts.wallet.clear();
+        client.eth.accounts.wallet.add(fromPrivateKey);
+        client.eth.defaultAccount = client.eth.accounts.wallet[0].address;
+    }
+    return client;
+};
+
+/**
+ * Sign XDC pending transaction from Tatum KMS
+ * @param tx pending transaction from KMS
+ * @param fromPrivateKey private key to sign transaction with.
+ * @param provider url of the XDC Server to connect to. If not set, default public server will be used.
+ * @returns transaction data to be broadcast to blockchain.
+ */
+export const signXdcKMSTransaction = async (tx: TransactionKMS, fromPrivateKey: string, provider?: string) => {
+    if (tx.chain !== Currency.XDC) {
+        throw Error('Unsupported chain.');
+    }
+    const client = getXdcClient(provider, fromPrivateKey);
+    const transactionConfig = JSON.parse(tx.serializedTransaction);
+    transactionConfig.gas = await client.eth.estimateGas(transactionConfig);
+    if (!transactionConfig.nonce) {
+        transactionConfig.nonce = await xdcGetTransactionsCount(client.eth.defaultAccount as string);
+    }
+    return (await client.eth.accounts.signTransaction(transactionConfig, fromPrivateKey as string)).rawTransaction as string;
+};
+
+/**
+ * Sign XDC Store data transaction with private keys locally. Nothing is broadcast to the blockchain.
+ * @param body content of the transaction to broadcast
+ * @param provider url of the XDC Server to connect to. If not set, default public server will be used.
+ * @returns transaction data to be broadcast to blockchain.
+ */
+export const prepareXdcStoreDataTransaction = async (body: CreateRecord, provider?: string) => {
+    await validateBody(body, CreateRecord);
+    const {
+        fromPrivateKey,
+        to,
+        ethFee,
+        data,
+        nonce,
+        signatureId
+    } = body;
+    const client = getXdcClient(provider, fromPrivateKey);
+    const address = to || client.eth.defaultAccount;
+    if (!address) {
+        throw new Error('Recipient must be provided.');
+    }
+    const addressNonce = nonce ? nonce : await xdcGetTransactionsCount(address);
+    const customFee = ethFee ? ethFee : {
+        gasLimit: `${data.length * 68 + 21000}`,
+        gasPrice: client.utils.fromWei(await xdcGetGasPriceInWei(), 'gwei'),
+    };
+
+    const tx: TransactionConfig = {
+        from: 0,
+        to: address.trim(),
+        value: '0',
+        gasPrice: customFee.gasPrice,
+        gas: customFee.gasLimit,
+        data: data ? (client.utils.isHex(data) ? client.utils.stringToHex(data) : client.utils.toHex(data)) : undefined,
```

Comment:
data is allwais here - no need t o ternary operator. Is XDC added in the CreateRecord.ts file?